misc/graphify.js

```diff
@@ -0,0 +1,124 @@
+var _ = require('lodash');
+var fs = require('fs');
+var geoip = require('geoip-lite');
+var crawler = require('../src/crawler.js');
+var normalizeIpp = crawler.normalizeIpp;
+var normalizePubKey = crawler.normalizePubKey;
+
+var rawCrawl = require('../' + process.argv.slice(2)[0]);
+
+function graphify(rawCrawl) {
+
+  var out = {nodes: [], links: [], iplinks: []}
+
+  // storage
+  var nodes = []
+  var links = {}
+  var indices = {}
+  var keyToIp = {}
+
+  // Collect all unique nodes and save their indices
+  for (var ipp in rawCrawl.data) {
+    indices[ipp] = nodes.length
+    if (nodes.length == 0) {
+      nodes.push({ipp: ipp, connections: 0, entry: 1})
+    } else {
+      nodes.push({ipp: ipp, connections: 0, entry: 0})
+    }
+    keyToIp[ipp]
+  }
+
+  // For each unique node
+  for(i in nodes) {
+    node = nodes[i];
+    ipp = node.ipp;
+    peers = rawCrawl.data[ipp].overlay.active;
+
+    // For each (not necessarily unique) peer
+    for (p_index in peers) {
+      peer = peers[p_index];
+
+      // Get ipp
+      try {
+        peer_ipp = normalizeIpp(peer.ip, peer.port);
+      } catch (err) {
+        peer_ipp = undefined;
+      }
+      peer_i = indices[peer_ipp];
+
+      // If it has an ipp
+      if (peer_ipp && peer_i !== undefined) {
+        peer_pk = normalizePubKey(peer.public_key);
+        peer_v = peer.version;
+        peer_t = peer.type;
+
+        // Fill out info in unique node list
+        peer_node = nodes[peer_i];
```

Comment:
Maybe don't call it a [unique node list](https://wiki.ripple.com/Unique_Node_List)